moveit_core/planning_interface/include/moveit/planning_interface/planning_response.h

```diff
@@ -37,33 +37,110 @@
 #pragma once
 
 #include <moveit/robot_trajectory/robot_trajectory.h>
-#include <moveit_msgs/MoveItErrorCodes.h>
+#include <moveit/utils/moveit_error_code.h>
 #include <moveit_msgs/MotionPlanResponse.h>
 #include <moveit_msgs/MotionPlanDetailedResponse.h>
 
 namespace planning_interface
 {
 struct MotionPlanResponse
 {
-  MotionPlanResponse() : planning_time_(0.0)
+  MotionPlanResponse() : trajectory_(trajectory), planning_time_(planning_time), error_code_(error_code)
+  {
+  }
+
+  MotionPlanResponse(const MotionPlanResponse& response)
+    : trajectory(response.trajectory)
+    , planning_time(response.planning_time)
+    , error_code(response.error_code)
+    , start_state(response.start_state)
+    , planner_id(response.planner_id)
+    , trajectory_(trajectory)
+    , planning_time_(planning_time)
+    , error_code_(error_code)
   {
   }
 
   void getMessage(moveit_msgs::MotionPlanResponse& msg) const;
 
-  robot_trajectory::RobotTrajectoryPtr trajectory_;
-  double planning_time_;
-  moveit_msgs::MoveItErrorCodes error_code_;
+  robot_trajectory::RobotTrajectoryPtr trajectory;
+  double planning_time;
+  moveit::core::MoveItErrorCode error_code;
+  moveit_msgs::RobotState start_state;
+  std::string planner_id;
+
+  [[deprecated("Use trajectory instead.")]] robot_trajectory::RobotTrajectoryPtr& trajectory_;
```

Comment:
@rhaschke Probably I am doing something wrong, but if I make these references const, the API breaks again. Do I miss  something here?